worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,265 @@
+        "setup_en.md",
+        "setup/en",
+        ["markustulliuscicero"]
+    )
+
+    tutorials = [setup_en]
+
+
+class JakAndDaxterWorld(World):
+    """
+    Jak and Daxter: The Precursor Legacy is a 2001 action platformer developed by Naughty Dog
+    for the PlayStation 2. The game follows the eponymous protagonists, a young boy named Jak
+    and his friend Daxter, who has been transformed into an ottsel. With the help of Samos
+    the Sage of Green Eco and his daughter Keira, the pair travel north in search of a cure for Daxter,
+    discovering artifacts created by an ancient race known as the Precursors along the way. When the
+    rogue sages Gol and Maia Acheron plan to flood the world with Dark Eco, they must stop their evil plan
+    and save the world.
+    """
+    # ID, name, version
+    game = jak1_name
+    required_client_version = (0, 4, 6)
+
+    # Options
+    settings: ClassVar[JakAndDaxterSettings]
+    options_dataclass = JakAndDaxterOptions
+    options: JakAndDaxterOptions
+
+    # Web world
+    web = JakAndDaxterWebWorld()
+
+    # Stored as {ID: Name} pairs, these must now be swapped to {Name: ID} pairs.
+    # Remember, the game ID and various offsets for each item type have already been calculated.
+    item_name_to_id = {item_table[k]: k for k in item_table}
+    location_name_to_id = {location_table[k]: k for k in location_table}
+    item_name_groups = {
+        "Power Cells": set(cell_item_table.values()),
+        "Scout Flies": set(scout_item_table.values()),
+        "Specials": set(special_item_table.values()),
+        "Moves": set(move_item_table.values()),
+        "Precursor Orbs": set(orb_item_table.values()),
+    }
+    location_name_groups = {
+        "Power Cells": set(cell_location_table.values()),
+        "Scout Flies": set(scout_location_table.values()),
+        "Specials": set(special_location_table.values()),
+        "Orb Caches": set(cache_location_table.values()),
+        "Precursor Orbs": set(orb_location_table.values()),
+        "Trades": {location_table[Cells.to_ap_id(k)] for k in
+                   {11, 12, 31, 32, 33, 96, 97, 98, 99, 13, 14, 34, 35, 100, 101}},
+    }
+
+    # Functions and Variables that are Options-driven, keep them as instance variables here so that we don't clog up
+    # the seed generation routines with options checking. So we set these once, and then just use them as needed.
+    can_trade: Callable[[CollectionState, int, Optional[int]], bool]
+    orb_bundle_size: int = 0
+    orb_bundle_item_name: str = ""
+
+    def generate_early(self) -> None:
+        # For the fairness of other players in a multiworld game, enforce some friendly limitations on our options,
+        # so we don't cause chaos during seed generation. These friendly limits should **guarantee** a successful gen.
+        if self.multiworld.players > 1:
+            from .Rules import enforce_multiplayer_limits
+            enforce_multiplayer_limits(self.options)
```

Comment:
Have you considered making a host.yaml setting to allow players to bypass these limits? Default should be off (disallowing disruptful options), of course.

Reply:
A host.yaml setting or a player option?

Reply:
A host.yaml setting. So that the host can specifically allow players to use a "worse" option (like if there's a sync between multiple Jak and Daxter games each with 1 orb orb-sanity on)

Reply:
```py
class JakAndDaxterSettings(settings.Group):
    class RootDirectory(settings.UserFolderPath):
        """Path to folder containing the ArchipelaGOAL mod executables (gk.exe and goalc.exe).
        Ensure this path contains forward slashes (/) only."""
        description = "ArchipelaGOAL Root Directory"

    class EnableUnfriendlyOptions(settings.Bool):
        """Allow unfriendly player options to be used in a multiplayer seed.
        This disables the 'friendly limits' checks for each player option."""
        description = "ArchipelaGOAL Enable Unfriendly Options"

    root_directory: RootDirectory = RootDirectory("%appdata%/OpenGOAL-Mods/archipelagoal")
    enable_unfriendly_options: EnableUnfriendlyOptions = False
```

Like so?

Reply:
Yeah, something like that o: